src/utils/analyzer/FunctionCallAnalyzer.ts

```diff
@@ -236,13 +237,30 @@ export class FunctionCallAnalyzer {
         const i = this.contractAnalyzer.interface.value
         const r = this.signatureResponse.value
         const functionHash = this.functionHash.value
-        if (functionHash !== null) {
+        const contractId = this.contractAnalyzer.contractId.value
+        const inputArgs = this.inputArgsOnly.value
+
+
+        if (functionHash !== null && inputArgs !== null) {
             if (i !== null) {
                 try {
-                    this.functionFragment.value = i.getFunction(functionHash)
+                    // this.functionFragment.value = i.getFunction(functionHash)
                     this.functionDecodingFailure.value = null
                     this.is4byteFunctionFragment.value = false
-                } catch (failure) {
+
+                    let currentFunctionFragment = i.getFunction(functionHash)
+                    // please refer to the ticket below for more information on this logic for redirectForToken(address,bytes) method on HTS System Contract 
+                    // https://github.com/hashgraph/hedera-mirror-node-explorer/issues/921
+                    if (
+                        currentFunctionFragment !== null &&
+                        contractId === HTS_PRECOMPILE_CONTRACT_ID &&
+                        functionHash === REDIRECT_FOR_TOKEN_FUNCTION_SIGHASH
+                    ) {
+                        currentFunctionFragment = resolveFunctionFragmentForHTSProxyContract(currentFunctionFragment, inputArgs)
+                    }
+
+                    this.functionFragment.value = currentFunctionFragment
+                } catch(failure) {
```

Comment:
Here we could use isRedirectToken() method and update control flow more explicitly:

```
         if (functionHash !== null && inputArgs !== null && contractId !== null) {
            if (i !== null) {
                try {
                    const ff = i.getFunction(functionHash)
                    if (isRedirectToken(contractId, functionHash) && ff !== null) {
                        this.functionFragment.value = resolveFunctionFragmentForHTSProxyContract(ff, inputArgs)
                        this.functionDecodingFailure.value = null
                        this.is4byteFunctionFragment.value = false
                    } else {
                        this.functionFragment.value = ff
                        this.functionDecodingFailure.value = null
                        this.is4byteFunctionFragment.value = false
                    }
                } catch(failure) {
                    this.functionFragment.value = null
                    this.functionDecodingFailure.value = failure
                    this.is4byteFunctionFragment.value = false
                }
            } else if (r !== null && r.results.length >= 1) {

```